examples/frigate/frigate.py

```diff
@@ -0,0 +1,143 @@
+# -*- coding: utf-8 -*-
+import dataclasses
+from datetime import datetime
+import typing as t
+
+from mqttwarn.context import RuntimeContext
+from mqttwarn.model import Service
+
+try:
+    import json
+except ImportError:
+    import simplejson as json
+
+
+@dataclasses.dataclass
+class FrigateEvent:
+    """
+    Manage inbound event data received from Frigate.
+    """
+    time: datetime
+    camera: str
+    label: str
+    current_zones: t.List[str]
+    entered_zones: t.List[str]
+
+    def f(self, value):
+        return [y.replace('_', ' ') for y in value]
+
+    @property
+    def current_zones_str(self):
+        return ', '.join(self.f(self.current_zones))
+
+    @property
+    def entered_zones_str(self):
+        return ', '.join(self.f(self.entered_zones))
+
+    def to_dict(self) -> t.Dict[str, str]:
+        return dataclasses.asdict(self)
+
+
+@dataclasses.dataclass
+class NtfyParameters:
+    """
+    Manage outbound parameter data for Apprise/Ntfy.
+    """
+    title: str
+    format: str
+    click: str
+    attach: str
+
+    def to_dict(self) -> t.Dict[str, str]:
+        return dataclasses.asdict(self)
+
+
+def frigate_events(topic, data, srv: Service):
+    """
+    mqttwarn transformation function which computes options to be submitted to Apprise/Ntfy.
+    """
+
+    # Acceptable hack to get attachment filename template from service configuration.
+    context: RuntimeContext = srv.mwcore["context"]
+    service_config = context.get_service_config("apprise-ntfy")
+    filename_template = service_config.get("filename_template")
+
+    # Decode JSON message.
+    after = json.loads(data['payload'])['after']
+
+    # Collect details from inbound Frigate event.
+    event = FrigateEvent(
+        time=datetime.fromtimestamp(after['frame_time']),
+        camera=after['camera'],
+        label=after['sub_label'] or after['label'],
+        current_zones=after['current_zones'],
+        entered_zones=after['entered_zones'],
+    )
+
+    # Interpolate event data into attachment filename template.
+    attach_filename = filename_template.format(**event.to_dict())
+
+    # Compute parameters for outbound Apprise / Ntfy URL.
+    ntfy_parameters = NtfyParameters(
+        title=f"{event.label} entered {event.entered_zones_str}",
+        format=f"In zones {event.current_zones_str} at {event.time}",
+        click=f"https://frigate/events?camera={event.camera}&label={event.label}&zone={event.entered_zones[0]}",
+        attach=attach_filename,
+    )
+    return ntfy_parameters.to_dict()
+
+
+def frigate_events_filter(topic, message, section, srv: Service):
+    """
+    mqttwarn filter function to only use Frigate events of type `new`.
```

Comment:
Technically this handles both `new` and `update`. There are 3 `type`s, of `new`, `update`, and `end`, so the filter will handle both `new` and `update` if it only excludes `end`.

Reply:
Thanks. Feel free to add any improvements on this detail. I am not so much into those aspects right now.

Reply:
```suggestion
    mqttwarn filter function to only use new events and important updates.
```

Reply:
Thanks. I've just added 29e93828b4a6 which addresses your suggestion.